src/web/collectors/CollectorFilter.jsx

```diff
@@ -0,0 +1,188 @@
+import React from 'react';
+import { Button } from 'react-bootstrap';
+import Immutable from 'immutable';
+
+import { TypeAheadInput } from 'components/common';
+
+const CollectorFilter = React.createClass({
+  propTypes: {
+    data: React.PropTypes.array,
```

Comment:
We should make these props required or provide default values to make ESLint happy.
https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/require-default-props.md